src/gen/conditions.rs

```diff
@@ -143,21 +144,29 @@ fn parse_args(
             // byte buffers (typically a 32 byte hash). We only pull out the
             // first element for now, but may support more in the future.
             // If we find anything else, that's still OK, since garbage is
-            // ignored.
-            if let Ok(c) = rest(a, c) {
-                // there was another item in the list
-                if let Ok(params) = first(a, c) {
-                    // the item was a cons-box, and params is the left-hand
-                    // side, the list element
-                    if let Ok(param) = first(a, params) {
-                        // pull out the first item (param)
-                        if let SExp::Atom(b) = a.sexp(param) {
-                            if a.buf(&b).len() <= 32 {
-                                return Ok(Condition::CreateCoin(puzzle_hash, amount, param));
-                            }
+            // ignored. (unless we're in mempool mode, and the STRICT_ARGS_COUNT
+            // flag is set)
+
+            // we always expect one more item, even if it's the zero-terminator
+            c = rest(a, c)?;
```

Comment:
This seems like a new error case. If it's intentional, that's okay. I guess it means a condition must be a list/can't be an atom?

Reply:
I don't think so.

`c` at the start of this match-block must be a cons-box where "first" is  puzzle-hash. Then `c` is updated to be the rest, which again must be a cons-box where "first" is the amount. Since `first()` already succeeded, `rest()` must also succeed without an error here. Whether we get back an atom, a nil terminator or another cons-box however, we don't know (yet).